### PR TITLE
fix doc about max_depth

### PR DESCRIPTION
**Pull request by CodingCat**

### CODE REVIEW

File: doc/parameter.rst [hunk 1]
```diff
@@ -64,8 +64,8 @@ Parameters for Tree Booster
 
 * ``max_depth`` [default=6]
 
-  - Maximum depth of a tree. Increasing this value will make the model more complex and more likely to overfit. 0 indicates no limit. Note that limit is required when ``grow_policy`` is set of ``depthwise``.
-  - range: [0,∞]
+  - Maximum depth of a tree. Increasing this value will make the model more complex and more likely to overfit. 0 is only accepted in ``lossguided`` growing policy when tree_method is set as ``hist`` and it indicates no limit on depth.
+  - range: [0,∞] (0 is only accepted in ``lossguided`` growing policy when tree_method is set as ``hist``)
```
Comment (thvasilo) [inline doc/parameter.rst]: This is a good opportunity to add a warning about setting this parameter too high, which is a usual cause for OOM errors from users. We should make it clear that every increase of 1 in this parameter doubles the size of the tree, and the memory usage of the tree/algorithm .

**Comment by CodingCat** [comment]
@hcho3 , travis-ci is broken again?

**Comment by hcho3** [comment]
@CodingCat This is rather frustrating. I might consider using other CI services for Mac testing, such as Circle CI.

**Comment by hcho3** [comment]
@CodingCat Okay, I found a simple fix for the broken Mac build (see dmlc/treelite#69), no need to migrate the CI. I'll file a new PR soon.

File: doc/parameter.rst [hunk 2]
```diff
@@ -64,8 +64,8 @@ Parameters for Tree Booster
 
 * ``max_depth`` [default=6]
 
-  - Maximum depth of a tree. Increasing this value will make the model more complex and more likely to overfit. 0 indicates no limit. Note that limit is required when ``grow_policy`` is set of ``depthwise``.
-  - range: [0,∞]
+  - Maximum depth of a tree. Increasing this value will make the model more complex and more likely to overfit. 0 is only accepted in ``lossguided`` growing policy when tree_method is set as ``hist`` and it indicates no limit on depth. Note: If you set this number too high, you may encounter Out-of-Memory (OOM) errors.
```
[the remark this answers is not in the document]
Comment (thvasilo) [reply doc/parameter.rst]: Agree with not using OOM here. Super-linear can cause confusion as well.
Could we just say that every increase by 1 in this parameter doubles the memory use of the method? That is roughly true no?

[the remark this answers is not in the document]
Comment (CodingCat) [reply doc/parameter.rst]: it's not necessarily double sized, as we may not split a node without enough gaining

and it is actually hard to measure as the memory consumption not only from model but also intermediate data

we can just say `XGBoost aggressively consumes memory when training a deep tree`

**Comment by hcho3** [comment]
Merged. Thanks!

**Pull request merged**